### ISSUE TITLE
Move ImageStream creation/handling from Pipelines to Operator

### ISSUE THREAD

**Issue by VannTen** [thoth-station/meteor-operator#126]
## High-level Goals
- Directly handles ImageStream inside the Operator
- No interactio between the Pipelines and the Openshift API
- Optionnaly: Use ImageStream to determine CNBi state (ImageAvailable)

## Proposal description
The ImageStream is created as part of the reconcilying loop and used as reconcile trigger

### Alternatives

Keep the current situation.

### Additional context
This was mentionned during CNBi meeting (9 Nov 2022)
Note that using ImageStream as the "impletementor" of CNBi rather than Pipeline might make more sense.

Consider the following scenarios:
- CNBi is created => Pipeline gets run => ImageStream created && image pushed.
=> What if the ImageSteam is deleted (by the user) ? In that case, a K8S operator should **reconcile**, and that means recreating the ImageStream
=> What if the underlying image is removed from the internal registry ? In that case, we'd need a new pipeline to rebuild the image
Note that both of these scenario do not imply a change of the PipelineRun resource, and will not (currently) trigger a reconcile request in the CNBi controller.

In a more theorical way, we should care about the result (= the ImageStream), not the process (= the PipelineRun) when doing level-based reconcile (the words of k8s docs).

## Acceptance Criteria

- [ ] CNBi controller create ImageStream with proper OwnerRefs
- (Optionnal, might be in another issue)[ ] Reconcile using ImageStream state

**Comment by VannTen** [comment]
/triage accepted 
(Based on last week CNBi meeting)

**Comment by VannTen** [comment]
https://github.com/kubernetes-sigs/controller-runtime/issues/1191#issuecomment-833058115

TL;DR: ImageStream CRDs are not easily availables. This is unfortunate because we use envtest, which needs to add CRDs used by the Operator to the test cluster, even if we don't test ImageStream functionnality specifically.
@goern We mentionned that in November, maybe we could talk with the Openshift product team (?) to see if they have suggestions ?

**Comment by goern** [comment]
It might also help to go thru the openshift-sme mailinglist, if it is still active ;) or drop by their slack.
